src/Symfony/Component/HttpFoundation/InputBag.php

```diff
@@ -36,29 +36,18 @@ public function get(string $key, $default = null)
         $value = parent::get($key, $this);
 
         if (null !== $value && $this !== $value && !is_scalar($value) && !(\is_object($value) && method_exists($value, '__toString'))) {
-            trigger_deprecation('symfony/http-foundation', '5.1', 'Retrieving a non-string value from "%s()" is deprecated, and will throw a "%s" exception in Symfony 6.0, use "%s::all()" instead.', __METHOD__, BadRequestException::class, __CLASS__);
+            trigger_deprecation('symfony/http-foundation', '5.1', 'Retrieving a non-string value from "%s()" is deprecated, and will throw a "%s" exception in Symfony 6.0, use "%s::all($key)" instead.', __METHOD__, BadRequestException::class, __CLASS__);
         }
 
         return $this === $value ? $default : $value;
     }
 
     /**
-     * Returns the inputs.
-     *
-     * @param string|null $key The name of the input to return or null to get them all
+     * {@inheritdoc}
      */
     public function all(string $key = null): array
     {
-        if (null === $key) {
-            return $this->parameters;
-        }
-
-        $value = $this->parameters[$key] ?? [];
-        if (!\is_array($value)) {
-            throw new BadRequestException(sprintf('Unexpected value for "%s" input, expecting "array", got "%s".', $key, get_debug_type($value)));
-        }
-
-        return $value;
+        return parent::all($key);
```

Comment:
I see that you tried making the behavior between the two bags more similar.. nice.. not sure if you were driven by my comment... but this is indeed nice..

Reply:
I don't remember exactly my flow of thoughts, but it might be :)